util/progress/progressui/init.go

```diff
@@ -0,0 +1,45 @@
+package progressui
+
+import (
+	"os"
+	"runtime"
+
+	"github.com/morikuni/aec"
+)
+
+var noColorIsSet bool
+var buildkitColorIsSet bool
+var colorRun aec.ANSI
+var colorCancel aec.ANSI
+var colorWarning aec.ANSI
+var colorError aec.ANSI
+var envColorString string
+
+func init() {
+	_, noColorIsSet = os.LookupEnv("NO_COLOR")
+	_, buildkitColorIsSet = os.LookupEnv("BUILDKIT_COLORS")
```

Comment:
I'm guessing the `NO_COLOR` option comes from https://no-color.org/? Might be useful to document that in a comment around it.

Would be good to respect it, but I think the right behavior might be to have `BUILDKIT_COLORS` override it if set (specific options should override general options imo)? I also think a `BUILDKIT_COLORS=none` option would be good, as a buildkit-specific way of forcing no colors - that could also extend into allowing multiple different "presets" in the future, if that was something we wanted to support.

Reply:
Adding a comment makes sense. I was going back and forth a bit on which variable should take precedence, and personally felt that `NO_COLOR` seemed like it should be the winner, as it seemed a bit more likely that an environment might have `BUILDKIT_COLORS` defined and then a specific script or pipeline would potentially disable it for one reason or another, but I am happy to change it if the general consensus is that `BUILDKIT_COLORS` should win.